src/Microsoft.DotNet.MSBuildSdkResolver/MSBuildSdkResolver.cs

```diff
@@ -292,17 +308,26 @@ private NETCoreSdkResolver.Result ResolveNETCoreSdkDirectory(SdkResolverContext
         {
             string globalJsonStartDir = Path.GetDirectoryName(context.SolutionFilePath ?? context.ProjectFilePath);
             var result = NETCoreSdkResolver.ResolveSdk(dotnetExeDir, globalJsonStartDir, _vsSettings.DisallowPrerelease());
+            result.GlobalJsonHonored = true;
 
-            if (result.ResolvedSdkDirectory != null
-                && result.GlobalJsonPath == null
-                && context.MSBuildVersion < GetMinimumMSBuildVersion(result.ResolvedSdkDirectory))
+            string mostCompatible = result.ResolvedSdkDirectory;
+            if (result.ResolvedSdkDirectory == null 
+                && result.GlobalJsonPath != null
+                && context.IsRunningInVisualStudio) // TODO this works when opening VS but once building in VS this is false and we get the old SDK resolution error.
```

Comment:
@rainersigwald @dsplaisted @wli3 during manual testing I ran into a problem here. It seems that `IsRunningInVisualStudio` is only true when we're opening VS, but once it's open and we're building inside it this flag is false and we're getting the old SDK resolution failure. Is there an alternative way I should[ set this flag from MSBuild](https://github.com/dotnet/msbuild/commit/699aeb65a984d7a5bdf71cbe4ab16b5e03528f25#diff-56a0887c6185e284af9f399c27a85c13R245) to get consistent behavior?

Reply:
This is solved by https://github.com/dotnet/msbuild/pull/5562, I'll update this PR once that change flows.

Reply:
You get the solution from Rainer?

Reply:
Yes, the solution is here: dotnet/msbuild#5562